test/jdk/com/sun/jdi/JdbStopInNotificationThreadTest.java

```diff
@@ -74,34 +73,70 @@ public void handleNotification(Notification n, Object hb) {
             }
         }, null, null);
 
-        Collection<Object[]> numbers = new LinkedList();
+        Collection<int[]> numbers = new LinkedList();
         long counter = 0;
+        System.out.println(tenuredGenPool.getName() + ": " + tenuredGenPool.getUsage());
         while (!done) {
-            numbers.add(new Object[1000]);
+            numbers.add(new int[1000]);
             counter++;
             if (counter % 1000 == 0) {
-                Thread.sleep(100);
+                System.gc();       // Encourage promotion into old/tenured generation
```

Comment:
I wonder if `fullGC` from `WhiteBox.java` can be used here -- it would make the intention to run full-gc explicit.

Reply:
Given that the test does not need it to reliably promote 100% into the oldgen on each call, only that allocated objects work there way there over time, I think the System.gc() is fine. It seems to be working well with all GC's, at least in the context of this test.